CollAction/Frontend/app/donation/DebitDetailsBox.tsx

```diff
@@ -0,0 +1,128 @@
+﻿import * as React from "react";
+import { IdealBankElement, IbanElement } from 'react-stripe-elements';
+
+interface IDebitDetailsBoxProps {
+    stripe: any;
+    amount: number;
+    userEmail: string;
+    userName: string;
+    isRecurring: boolean;
+}
+
+interface IDebitDetailsBoxState {
+    showError: boolean;
+    error: string;
+}
+
+export default class DebitDetailsBox extends React.Component<IDebitDetailsBoxProps, IDebitDetailsBoxState> {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showError: false,
+            error: ""
+        };
+    }
+
+    submitPayment(event: React.FormEvent<HTMLFormElement>) : Promise<void> {
```

Comment:
No need for the : Promise<void> return type here; typescript infers this.

Reply:
I know typescript "interferes" it, but I like to at least specify types for functions.